### ISSUE TITLE
Select version of jQuery for 4.9.0 release

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern#1941]
We've got two jQuery versions in play in https://github.com/textpattern/textpattern/tree/dev/textpattern/vendors/jquery/jquery - presumably we can pick one and save ~80kB prior to 4.9.0 release.

Tagging https://github.com/textpattern/textpattern/issues/1896

**Comment by phiw13** [comment]
I think you can safely delete the `jquery-4.0.0-beta.min.js` file.

* it is outdated (that is beta1, beta2 has since been released)
* it is not used
* if Oleg (@bloatware ) wants to test more with jq4, he’ll use the most recent beta-gama-delta release at that moment.

**Comment by petecooper** [comment]
I've added beta2 via https://github.com/textpattern/textpattern/commit/7fc9d035b27694858418f5e348189659b5273731

We're aiming for a December release, so we may have the GA release by then.

**Comment by phiw13** [comment]
you removed beta 1, but now TXP-dev (diagnostics) complains, as it is included in the `checksums` file.

(no don’t hurry to fix that one…)

**Comment by petecooper** [comment]
Nuts. Thanks for the report.

**Comment by Bloke** [comment]
If the jQuery team get their release done, and testing goes well here, we can just swap it out in a month or two.

**Comment by phiw13** [comment]
@bloatware 

What issues did you encounter? I spend an hour or so with jq4-beta2, Txp-dev latest and some more strict CSP rules without running into anything going wrong. Writing, saving, editing, saving an article as draft and then live, checking the preview widget, uploading Images (dedicated panel and com_article_image) & Files, etc etc.

**Comment by bloatware** [comment]
@phiw13 there is no issue on my localhost. I have only observed it on the demo server, but do not remember what it was.

Some plugins might need to be updated too if we switch to jq4.

**Comment by petecooper** [comment]
I'll be around for CSP tweaks on the demo server if we can shoot for an upgrade there.

**Comment by petecooper** [comment]
>Some plugins might need to be updated too if we switch to jq4.

With that in mind, my instinct says to leave Textpattern 4.9 with the more established jQuery 3, given there may be some plugin breakages. Textpattern 5 with jQuery 4 might make more sense, and not rock the developer boat quite so much.

jQuery 3.0.0 had an alpha (13 July 2015), a beta (14 January 2016), a release candidate (20 May 2016), and then went to GA (9 June 2016). There are an increasing number of npm downloads for jQuery, so its popularity isn't going away.

**Comment by petecooper** [comment]
@bloatware @Bloke Checking in for advice on jQuery version. What version do you want to include in the 4.9.0 distribution?

**Comment by bloatware** [comment]
Unless they release jQuery 4rc in a near future, do we really have choice? Locally, jQ4beta2 works for me, fwiw.

**Comment by phiw13** [comment]
One serious issue with JQ4beta2: on small  screen devices the menu system fails to open: the top level button is not functional (you can reproduce with  a desktop browser: enter responsive design mode (dev. tools) and try clicking the menu.

This is a bit of an old (Bootstrap 3) menu script that has worked very well and is compact without any dependencies other that Jquery. Source is here I think: https://github.com/twbs/bootstrap/blob/v3-dev/js/dropdown.js


Safari -> “TypeError: No default value” main.js:8:2988
Firefox -> “Uncaught TypeError: can't convert Object to string” main.js:8:2983

the offending line as seen in Safari, un-minified view of the js file
`!e && n.toggle && /show|hide/.test(a) && (n.toggle = !1),`

in this block

```
function s(a) {
        return this.each(function() {
            var t = i(this),
                e = t.data("bs.collapse"),
                n = i.extend({}, o.DEFAULTS, t.data(), "object" == typeof a && a);
            !e && n.toggle && /show|hide/.test(a) && (n.toggle = !1),
            e || t.data("bs.collapse", e = new o(this, n)),
            "string" == typeof a && e[a]()
        })
    }
```


This Sandspace mockup page: https://dev.l-c-n.com/txp-4.9/txp-content-write-jq4beta.html runs JQ4 and reproduces the issue.

I have no idea if that script can be hacked to fix it.

**Comment by bloatware** [comment]
Yes, indeed. While this issue can be patched, it looks risky to switch to jQuery 4 in 4.9...

**Comment by petecooper** [comment]
>it looks risky to switch to jQuery 4 in 4.9...

I'm inclined to agree. At least jQuery 4 will (ideally) be fairly settled for Textpattern 5. We could merge it into a patch release easily enough.

@Bloke Are you comfortable staying with jQuery 3.current for Textpattern 4.9.0?

**Comment by petecooper** [comment]
Based on above, I've tentatively set v5.0 milestone for https://github.com/textpattern/textpattern/issues/1896

@Bloke Sorry to nag - all ok with jQuery 3.x in Textpattern 4.9.0?

**Comment by petecooper** [comment]
Boarding call for passenger @Bloke - given all the other moving parts in 4.9.0, thumbs up or down for staying with jQuery 3.x?

**Comment by phiw13** [comment]
> Boarding call for passenger [@Bloke](https://github.com/Bloke) - given all the other moving parts in 4.9.0, thumbs up or down for staying with jQuery 3.x?

Are you OK with the menu system not working on small(er) screen devices with JQuery 4 – still in beta?

**Comment by Bloke** [comment]
> Personally, no. I'm +1 for keeping jQuery 3.x in Textpattern 4.9.0.

Ditto.

**Comment by phiw13** [comment]
Everybody agree! :-)

**Comment by petecooper** [comment]
Closing for now. Thanks, everyone.
